### PR TITLE
Implement NSHTTPURLResponse.localizedStringForStatusCode()

### PR DESCRIPTION
**Pull request by ianpartridge**
Implement according to the IANA registry code descriptions at http://www.iana.org/assignments/http-status-codes/http-status-codes.xhtml

Match the behaviour of Obj-C Foundation for unknown status codes.

This does not do translation, but neither does the Obj-C version of Foundation.  If required, that could be added in future.

### CODE REVIEW

File: Foundation/NSURLResponse.swift [hunk 1]
```diff
@@ -176,6 +176,74 @@ public class NSHTTPURLResponse : NSURLResponse {
...
+        case 503: return "Service Unavailable"
+        case 504: return "Gateway Timeout"
+        case 505: return "HTTP Version Not Supported"
+        case 506: return "Variant Also Negotiates"
+        case 507: return "Insufficient Storage"
+        case 508: return "Loop Detected"
+        case 510: return "Not Extended"
+        case 511: return "Network Authentication Required"
+        case 500...599: return "Server Error"
+        default: return "Server Error"
```
Comment (phausler) [inline Foundation/NSURLResponse.swift]:
```swift
for n in 0..<600 {
    let s = NSHTTPURLResponse.localizedString(forStatusCode: n)
    print("case \(n): return \"\(s)\"")
}
```

prints:

```swift
case 0: return "server error"
case 1: return "server error"
case 2: return "server error"
case 3: return "server error"
case 4: return "server error"
case 5: return "server error"
case 6: return "server error"
case 7: return "server error"
case 8: return "server error"
case 9: return "server error"
case 10: return "server error"
case 11: return "server error"
case 12: return "server error"
case 13: return "server error"
case 14: return "server error"
case 15: return "server error"
case 16: return "server error"
case 17: return "server error"
case 18: return "server error"
case 19: return "server error"
case 20: return "server error"
case 21: return "server error"
case 22: return "server error"
case 23: return "server error"
case 24: return "server error"
case 25: return "server error"
case 26: return "server error"
case 27: return "server error"
case 28: return "server error"
case 29: return "server error"
case 30: return "server error"
case 31: return "server error"
case 32: return "server error"
case 33: return "server error"
case 34: return "server error"
case 35: return "server error"
case 36: return "server error"
case 37: return "server error"
case 38: return "server error"
case 39: return "server error"
case 40: return "server error"
case 41: return "server error"
case 42: return "server error"
case 43: return "server error"
case 44: return "server error"
case 45: return "server error"
case 46: return "server error"
case 47: return "server error"
case 48: return "server error"
case 49: return "server error"
case 50: return "server error"
case 51: return "server error"
case 52: return "server error"
case 53: return "server error"
case 54: return "server error"
case 55: return "server error"
case 56: return "server error"
case 57: return "server error"
case 58: return "server error"
case 59: return "server error"
case 60: return "server error"
case 61: return "server error"
case 62: return "server error"
case 63: return "server error"
case 64: return "server error"
case 65: return "server error"
case 66: return "server error"
case 67: return "server error"
case 68: return "server error"
case 69: return "server error"
case 70: return "server error"
case 71: return "server error"
case 72: return "server error"
case 73: return "server error"
case 74: return "server error"
case 75: return "server error"
case 76: return "server error"
case 77: return "server error"
case 78: return "server error"
case 79: return "server error"
case 80: return "server error"
case 81: return "server error"
case 82: return "server error"
case 83: return "server error"
case 84: return "server error"
case 85: return "server error"
case 86: return "server error"
case 87: return "server error"
case 88: return "server error"
case 89: return "server error"
case 90: return "server error"
case 91: return "server error"
case 92: return "server error"
case 93: return "server error"
case 94: return "server error"
case 95: return "server error"
case 96: return "server error"
case 97: return "server error"
case 98: return "server error"
case 99: return "server error"
case 100: return "continue"
case 101: return "switching protocols"
case 102: return "informational"
case 103: return "informational"
case 104: return "informational"
case 105: return "informational"
case 106: return "informational"
case 107: return "informational"
case 108: return "informational"
case 109: return "informational"
case 110: return "informational"
case 111: return "informational"
case 112: return "informational"
case 113: return "informational"
case 114: return "informational"
case 115: return "informational"
case 116: return "informational"
case 117: return "informational"
case 118: return "informational"
case 119: return "informational"
case 120: return "informational"
case 121: return "informational"
case 122: return "informational"
case 123: return "informational"
case 124: return "informational"
case 125: return "informational"
case 126: return "informational"
case 127: return "informational"
case 128: return "informational"
case 129: return "informational"
case 130: return "informational"
case 131: return "informational"
case 132: return "informational"
case 133: return "informational"
case 134: return "informational"
case 135: return "informational"
case 136: return "informational"
case 137: return "informational"
case 138: return "informational"
case 139: return "informational"
case 140: return "informational"
case 141: return "informational"
case 142: return "informational"
case 143: return "informational"
case 144: return "informational"
case 145: return "informational"
case 146: return "informational"
case 147: return "informational"
case 148: return "informational"
case 149: return "informational"
case 150: return "informational"
case 151: return "informational"
case 152: return "informational"
case 153: return "informational"
case 154: return "informational"
case 155: return "informational"
case 156: return "informational"
case 157: return "informational"
case 158: return "informational"
case 159: return "informational"
case 160: return "informational"
case 161: return "informational"
case 162: return "informational"
case 163: return "informational"
case 164: return "informational"
case 165: return "informational"
case 166: return "informational"
case 167: return "informational"
case 168: return "informational"
case 169: return "informational"
case 170: return "informational"
case 171: return "informational"
case 172: return "informational"
case 173: return "informational"
case 174: return "informational"
case 175: return "informational"
case 176: return "informational"
case 177: return "informational"
case 178: return "informational"
case 179: return "informational"
case 180: return "informational"
case 181: return "informational"
case 182: return "informational"
case 183: return "informational"
case 184: return "informational"
case 185: return "informational"
case 186: return "informational"
case 187: return "informational"
case 188: return "informational"
case 189: return "informational"
case 190: return "informational"
case 191: return "informational"
case 192: return "informational"
case 193: return "informational"
case 194: return "informational"
case 195: return "informational"
case 196: return "informational"
case 197: return "informational"
case 198: return "informational"
case 199: return "informational"
case 200: return "no error"
case 201: return "created"
case 202: return "accepted"
case 203: return "non-authoritative information"
case 204: return "no content"
case 205: return "reset content"
case 206: return "partial content"
case 207: return "success"
case 208: return "success"
case 209: return "success"
case 210: return "success"
case 211: return "success"
case 212: return "success"
case 213: return "success"
case 214: return "success"
case 215: return "success"
case 216: return "success"
case 217: return "success"
case 218: return "success"
case 219: return "success"
case 220: return "success"
case 221: return "success"
case 222: return "success"
case 223: return "success"
case 224: return "success"
case 225: return "success"
case 226: return "success"
case 227: return "success"
case 228: return "success"
case 229: return "success"
case 230: return "success"
case 231: return "success"
case 232: return "success"
case 233: return "success"
case 234: return "success"
case 235: return "success"
case 236: return "success"
case 237: return "success"
case 238: return "success"
case 239: return "success"
case 240: return "success"
case 241: return "success"
case 242: return "success"
case 243: return "success"
case 244: return "success"
case 245: return "success"
case 246: return "success"
case 247: return "success"
case 248: return "success"
case 249: return "success"
case 250: return "success"
case 251: return "success"
case 252: return "success"
case 253: return "success"
case 254: return "success"
case 255: return "success"
case 256: return "success"
case 257: return "success"
case 258: return "success"
case 259: return "success"
case 260: return "success"
case 261: return "success"
case 262: return "success"
case 263: return "success"
case 264: return "success"
case 265: return "success"
case 266: return "success"
case 267: return "success"
case 268: return "success"
case 269: return "success"
case 270: return "success"
case 271: return "success"
case 272: return "success"
case 273: return "success"
case 274: return "success"
case 275: return "success"
case 276: return "success"
case 277: return "success"
case 278: return "success"
case 279: return "success"
case 280: return "success"
case 281: return "success"
case 282: return "success"
case 283: return "success"
case 284: return "success"
case 285: return "success"
case 286: return "success"
case 287: return "success"
case 288: return "success"
case 289: return "success"
case 290: return "success"
case 291: return "success"
case 292: return "success"
case 293: return "success"
case 294: return "success"
case 295: return "success"
case 296: return "success"
case 297: return "success"
case 298: return "success"
case 299: return "success"
case 300: return "multiple choices"
case 301: return "moved permanently"
case 302: return "found"
case 303: return "see other"
case 304: return "not modified"
case 305: return "needs proxy"
case 306: return "redirected"
case 307: return "temporarily redirected"
case 308: return "redirected"
case 309: return "redirected"
case 310: return "redirected"
case 311: return "redirected"
case 312: return "redirected"
case 313: return "redirected"
case 314: return "redirected"
case 315: return "redirected"
case 316: return "redirected"
case 317: return "redirected"
case 318: return "redirected"
case 319: return "redirected"
case 320: return "redirected"
case 321: return "redirected"
case 322: return "redirected"
case 323: return "redirected"
case 324: return "redirected"
case 325: return "redirected"
case 326: return "redirected"
case 327: return "redirected"
case 328: return "redirected"
case 329: return "redirected"
case 330: return "redirected"
case 331: return "redirected"
case 332: return "redirected"
case 333: return "redirected"
case 334: return "redirected"
case 335: return "redirected"
case 336: return "redirected"
case 337: return "redirected"
case 338: return "redirected"
case 339: return "redirected"
case 340: return "redirected"
case 341: return "redirected"
case 342: return "redirected"
case 343: return "redirected"
case 344: return "redirected"
case 345: return "redirected"
case 346: return "redirected"
case 347: return "redirected"
case 348: return "redirected"
case 349: return "redirected"
case 350: return "redirected"
case 351: return "redirected"
case 352: return "redirected"
case 353: return "redirected"
case 354: return "redirected"
case 355: return "redirected"
case 356: return "redirected"
case 357: return "redirected"
case 358: return "redirected"
case 359: return "redirected"
case 360: return "redirected"
case 361: return "redirected"
case 362: return "redirected"
case 363: return "redirected"
case 364: return "redirected"
case 365: return "redirected"
case 366: return "redirected"
case 367: return "redirected"
case 368: return "redirected"
case 369: return "redirected"
case 370: return "redirected"
case 371: return "redirected"
case 372: return "redirected"
case 373: return "redirected"
case 374: return "redirected"
case 375: return "redirected"
case 376: return "redirected"
case 377: return "redirected"
case 378: return "redirected"
case 379: return "redirected"
case 380: return "redirected"
case 381: return "redirected"
case 382: return "redirected"
case 383: return "redirected"
case 384: return "redirected"
case 385: return "redirected"
case 386: return "redirected"
case 387: return "redirected"
case 388: return "redirected"
case 389: return "redirected"
case 390: return "redirected"
case 391: return "redirected"
case 392: return "redirected"
case 393: return "redirected"
case 394: return "redirected"
case 395: return "redirected"
case 396: return "redirected"
case 397: return "redirected"
case 398: return "redirected"
case 399: return "redirected"
case 400: return "bad request"
case 401: return "unauthorized"
case 402: return "payment required"
case 403: return "forbidden"
case 404: return "not found"
case 405: return "method not allowed"
case 406: return "unacceptable"
case 407: return "proxy authentication required"
case 408: return "request timed out"
case 409: return "conflict"
case 410: return "no longer exists"
case 411: return "length required"
case 412: return "precondition failed"
case 413: return "request too large"
case 414: return "requested URL too long"
case 415: return "unsupported media type"
case 416: return "requested range not satisfiable"
case 417: return "expectation failed"
case 418: return "client error"
case 419: return "client error"
case 420: return "client error"
case 421: return "client error"
case 422: return "client error"
case 423: return "client error"
case 424: return "client error"
case 425: return "client error"
case 426: return "client error"
case 427: return "client error"
case 428: return "client error"
case 429: return "client error"
case 430: return "client error"
case 431: return "client error"
case 432: return "client error"
case 433: return "client error"
case 434: return "client error"
case 435: return "client error"
case 436: return "client error"
case 437: return "client error"
case 438: return "client error"
case 439: return "client error"
case 440: return "client error"
case 441: return "client error"
case 442: return "client error"
case 443: return "client error"
case 444: return "client error"
case 445: return "client error"
case 446: return "client error"
case 447: return "client error"
case 448: return "client error"
case 449: return "client error"
case 450: return "client error"
case 451: return "client error"
case 452: return "client error"
case 453: return "client error"
case 454: return "client error"
case 455: return "client error"
case 456: return "client error"
case 457: return "client error"
case 458: return "client error"
case 459: return "client error"
case 460: return "client error"
case 461: return "client error"
case 462: return "client error"
case 463: return "client error"
case 464: return "client error"
case 465: return "client error"
case 466: return "client error"
case 467: return "client error"
case 468: return "client error"
case 469: return "client error"
case 470: return "client error"
case 471: return "client error"
case 472: return "client error"
case 473: return "client error"
case 474: return "client error"
case 475: return "client error"
case 476: return "client error"
case 477: return "client error"
case 478: return "client error"
case 479: return "client error"
case 480: return "client error"
case 481: return "client error"
case 482: return "client error"
case 483: return "client error"
case 484: return "client error"
case 485: return "client error"
case 486: return "client error"
case 487: return "client error"
case 488: return "client error"
case 489: return "client error"
case 490: return "client error"
case 491: return "client error"
case 492: return "client error"
case 493: return "client error"
case 494: return "client error"
case 495: return "client error"
case 496: return "client error"
case 497: return "client error"
case 498: return "client error"
case 499: return "client error"
case 500: return "internal server error"
case 501: return "unimplemented"
case 502: return "bad gateway"
case 503: return "service unavailable"
case 504: return "gateway timed out"
case 505: return "unsupported version"
case 506: return "server error"
case 507: return "server error"
case 508: return "server error"
case 509: return "server error"
case 510: return "server error"
case 511: return "server error"
case 512: return "server error"
case 513: return "server error"
case 514: return "server error"
case 515: return "server error"
case 516: return "server error"
case 517: return "server error"
case 518: return "server error"
case 519: return "server error"
case 520: return "server error"
case 521: return "server error"
case 522: return "server error"
case 523: return "server error"
case 524: return "server error"
case 525: return "server error"
case 526: return "server error"
case 527: return "server error"
case 528: return "server error"
case 529: return "server error"
case 530: return "server error"
case 531: return "server error"
case 532: return "server error"
case 533: return "server error"
case 534: return "server error"
case 535: return "server error"
case 536: return "server error"
case 537: return "server error"
case 538: return "server error"
case 539: return "server error"
case 540: return "server error"
case 541: return "server error"
case 542: return "server error"
case 543: return "server error"
case 544: return "server error"
case 545: return "server error"
case 546: return "server error"
case 547: return "server error"
case 548: return "server error"
case 549: return "server error"
case 550: return "server error"
case 551: return "server error"
case 552: return "server error"
case 553: return "server error"
case 554: return "server error"
case 555: return "server error"
case 556: return "server error"
case 557: return "server error"
case 558: return "server error"
case 559: return "server error"
case 560: return "server error"
case 561: return "server error"
case 562: return "server error"
case 563: return "server error"
case 564: return "server error"
case 565: return "server error"
case 566: return "server error"
case 567: return "server error"
case 568: return "server error"
case 569: return "server error"
case 570: return "server error"
case 571: return "server error"
case 572: return "server error"
case 573: return "server error"
case 574: return "server error"
case 575: return "server error"
case 576: return "server error"
case 577: return "server error"
case 578: return "server error"
case 579: return "server error"
case 580: return "server error"
case 581: return "server error"
case 582: return "server error"
case 583: return "server error"
case 584: return "server error"
case 585: return "server error"
case 586: return "server error"
case 587: return "server error"
case 588: return "server error"
case 589: return "server error"
case 590: return "server error"
case 591: return "server error"
case 592: return "server error"
case 593: return "server error"
case 594: return "server error"
case 595: return "server error"
case 596: return "server error"
case 597: return "server error"
case 598: return "server error"
case 599: return "server error"
```

Comment (ianpartridge) [inline Foundation/NSURLResponse.swift]: If IANA recognise 418 I'll be happy to add it :-)

Do you want the code to exactly match the Obj-C Foundation output?

**Comment by phausler** [comment]
I would need to verify the localizations that this supports; but overall this looks like a decent starting point

**Comment by phausler** [comment]
@swift-ci Please test

**Comment by ianpartridge** [comment]
Hi @phausler - I rebased this against head so the conflicts are fixed.  Merge if you are happy? 😀

**Comment by parkera** [comment]
It looks like we ended up with more conflicts. =(

**Comment by ianpartridge** [comment]
Sigh.  Rebased against HEAD again.  Any chance of a quick CI and merge @parkera / @phausler ?

**Pull request merged**